File: helloWorld/businessLayer-Scala/src/main/scala/uk/gov/ons/collection/businesslayer/entity/Person.scala
package uk.gov.ons.collection.businesslayer.entity


class Person(_id: Long, _firstname: String, _surname: String) {
  var id = _id
  var firstname =  _firstname
  var surname = _surname

  def formatName (name: String) : String = {
    if (name.length == 0) {
      return  name
    }
    return name.substring(0,1).toUpperCase() + name.substring(1).toLowerCase()
  }

  def stripAll(s: String, bad: String): String = {

    @scala.annotation.tailrec def start(n: Int): String =
      if (n == s.length) ""
      else if (bad.indexOf(s.charAt(n)) < 0) end(n, s.length)
      else start(1 + n)

    @scala.annotation.tailrec def end(a: Int, n: Int): String =
      if (n <= a) s.substring(a, n)
      else if (bad.indexOf(s.charAt(n - 1)) < 0) s.substring(a, n)
      else end(a, n - 1)

    start(0)
  }

  def getFormattedFirstName : String = {
    return formatName(stripAll(firstname.toString," ")).toLowerCase.capitalize
  }

  def getFormattedLastName : String = {
    return formatName(stripAll(surname.toString," ")).toLowerCase.capitalize
  }

  override def toString = {
    s"${id} ${firstname} ${surname}"
  }
}

File: helloWorld/businessLayer/src/main/java/uk/gov/ons/collection/entity/Person.java
package uk.gov.ons.collection.entity;

public class Person{
    private final Integer id;
    private final String firstname;
    private final String surname;

    public Person(Integer id, String firstname, String surname) {
        this.id = id;
        this.firstname = (firstname == null) ? "" : firstname; // Assign empty string in case of a null
        this.surname = (surname == null) ? "" : surname;       // Assign empty string in case of a null
    }

    public Integer getId() {
        return id;
    }

    public String getFirstname() {
        return firstname;
    }

    public String getLastname(){
        return surname;
    }

    private String formatName(String name){
        if (name.length() == 0) {
            return name;
        }
        return name.substring(0, 1).toUpperCase() + name.substring(1).toLowerCase();
    }

    public String getFormattedFirstname(){
        return formatName(firstname.trim());
    }

    public String getFormattedLastname(){
        return formatName(surname.trim());
    }

    @Override
    public String toString() {
        return this.id.toString() + " " + this.firstname + " " + this.surname;
    }
}
